HikVisionSource/README.md

```diff
@@ -0,0 +1,217 @@
+3. Navigate to `<repo location>/vantiq-extension-sources/HikVisionSource/build/distributions`. The zip and tar files both contain the same files, so choose whichever you prefer.
+4. Uncompress the file in the location that you would like to install the program.
+5. Run `<install location>/HikVisionSource/bin/HikVisionSource` with a local server.config file or specifying the [server config file](#serverConfig) as the first argument.
+
+## Logging
+
+To change the logging settings, edit the logging config file `<install location>/HikVisionSource/src/main/resources/log4j2.xml` , 
+which is an [Apache Log4j configuration file.](https://logging.apache.org/log4j/2.x/manual/configuration.html). The logger 
+name for each class is the class's fully qualified class name, *e.g.* "io.vantiq.extjsdk. ExtensionWebSocketClient".  
+
+## Server Config File
+
+The server config file is written as `property=value` , with each property on its
+own line. The following is an example of a valid server.config file:
+
+``` 
+authToken=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+sources=HikVision1
+targetServer=https://dev.vantiq.com/
+```
+
+### Vantiq Options
+
+* **authToken**: Required. The authentication token to connect with. These can be obtained from the namespace admin.
+* **sources**: Required. A comma separated list of the sources to which you wish to connect. Any whitespace will be removed when read.
+* **targetServer**: Required. The Vantiq server hosting the sources.
+
+# Setting Up Your Vantiq Environment<a name="vantiq" id="vantiq"></a>
+
+An understanding of the VANTIQ Extension Source SDK is assumed. Please read the [Extension Source README.md](../README.md) for more information.
+
+In order to incorporate this Extension Source, you will need to create the Source Implementation in the Vantiq system.
+
+## Source Implementation
+
+When creating a HikVision Extension source, you must first create the source implementation. This is done by using the [*HikVisionImpl.json*](src/test/resources/HikVisionImpl.json) file found in src/test/resources/HikVisionImpl.json. To make the source type known to Vantiq, use the vantiq cli command
+
+```
+vantiq -s <profileName> load sourceimpls <fileName>
+```
+
+where <profileName> is replaced by the Vantiq profile name, and <fileName> is the file to be loaded.
+
+(You can, of course, change the source implementation name from that provided in this definition file.)
+
+That completed, you will need to create the Source in the Vantiq system.
+
+## Source Configuration
+
+To set up the Source in the Vantiq system, you will need to add a Source to your project.  Make sure you have properly added a Source Implementation to Vantiq. Once this is complete, you can select HikVision (or whatever you named your Source Definition) as the Source Type.
+You will then need to fill out the Source Configuration Document.
+
+The Configuration document may look similar to the following example:
```

Comment:
This line should move down to just before the example.

Reply:
done